lib/tasks/annual_fund_impact_metrics_activities.rake

```diff
@@ -0,0 +1,42 @@
+require "csv-safe"
+
+namespace :activities do
+  task :annual_fund_impact_metrics, [:output_csv_path] => :environment do |_task, args|
+    desc "Generates a CSV file of Activities that will be used to pre-populate an annual
+          collaborative spreadsheet of fund impact metrics"
+
+    csv_path = args[:output_csv_path] || "tmp/annual_fund_impact_metrics.csv"
+
+    CSVSafe.open(csv_path, "wb") do |csv|
+      activities = Activity
+        .joins(:organisation)
+        .includes(:organisation, :actuals)
+        .where.not(programme_status: ["delivery", "agreement_in_place", "open_for_applications", "stopped"])
+        .order("organisations.name, programme_status")
+
+      # We want to exclude Activities that were completed more than two years ago. We have discussed
+      # this and determined that the best way to do this is to exclude those that have no Actuals
+      # reported in the last two years (including if there are no Actuals at all).
+      active_activities = activities.reject do |activity|
+        completed = activity.programme_status == "completed"
```

Comment:
Could we exit early from this block if the activity is not completed?

Maybe if we extract the block into a method that returns true or false, and it can return false quickly if the activity isn't completed, without needing to fetch and examine the actuals.

Not sure if this is premature optimisation, but worth checking.